src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -37,7 +37,9 @@ void addChannel(SnowflakeStreamingIngestChannelInternal<T> channel) {
         channels.put(channel.getName(), channel);
     // Invalidate old channel if it exits to block new inserts and return error to users earlier
     if (oldChannel != null) {
-      oldChannel.invalidate("removed from cache");
+      String errorMessage =
```

Comment:
Removing from cache is not an error, maybe let's call it invalidationReason or invalidationCause?